daemon/src/main/scala/io/gearpump/cluster/scheduler/PriorityScheduler.scala

```diff
@@ -70,56 +69,78 @@ class PriorityScheduler extends Scheduler{
             scheduleLater = scheduleLater :+ PendingRequest(appId, appMaster, request, timeStamp)
           }
         case SPECIFICWORKER =>
-          if (resourcesSnapShot.contains(request.workerId)) {
-            val (worker, availableResource) = resourcesSnapShot.get(request.workerId).get
-            if (availableResource > request.resource) {
-              appMaster ! ResourceAllocated(Array(ResourceAllocation(request.resource, worker, request.workerId)))
-              allocated = allocated  + request.resource
-              resourcesSnapShot.update(request.workerId, (worker, availableResource - request.resource))
-            }
+          val workerAndResource = resourcesSnapShot.get(request.workerId)
```

Comment:
There was a bug before the change. If worker's remaining resource is not larger than the required resource, the request is not put back.